src/os/bluestore/BlueFS.cc

```diff
@@ -2216,6 +2223,15 @@ int BlueFS::open_for_read(
   return 0;
 }
 
+void BlueFS::_close_reader(FileReader *h)
+{
+  FileRef file = h->file;
+  delete h;
+  if (file->deleted && file->num_readers == 0) {
```

Comment:
May be move this check inside _finish_unlink and call it unconditionally here and above.

Reply:
good idea, pushed a cleanup (that also asserts num_writers == 0)